litmus-portal/backend/graphql-server/graph/schema.graphqls

```diff
@@ -2,20 +2,66 @@
 #
 # https://gqlgen.com/getting-started/
 
-input clusterInput{
-  id: Int!
-  data: String!
+type Cluster {
+  cluster_id: ID!
+  project_id: ID!
+  cluster_name: String!
+  description: String!
+  platform_name: String!
+  access_key: String!
+  is_registered: Boolean!
+  is_active: Boolean!
+  updated_at: String!
+  created_at: String!
+  cluster_type: String!
 }
 
-type clusterResponse{
-  id: Int!
-  data: String!
+input ClusterInput{
+  cluster_name: String!
+  description: String!
+  platform_name: String!
+  project_id: ID!
+  cluster_type: String!
 }
 
-type Mutation {
-  createCluster(input: clusterInput!): clusterResponse!
+type ClusterEvent {
+  event_id: ID!
+  event_type: String!
+  event_name: String!
+  description: String!
+  cluster: Cluster!
 }
 
-type Subscription {
-  clusterSubscription: clusterResponse!
+type ClusterAction {
+    project_id: ID!
+    action:String!
+}
+
+input ClusterActionInput {
+    cluster_id: ID!
+    action:String!
+}
+
+input ClusterEventInput{
+  event_name: String!
+  description: String!
+  cluster_id: String!
+  access_key:String!
+}
+
+input ClusterIdentity{
+  cluster_id: String!
+  access_key: String!
+}
+
+type Mutation{
+    userClusterReg(clusterInput:ClusterInput!):String!
+    clusterConfirm(identity:ClusterIdentity!):String!
+    newClusterEvent(clusterEvent:ClusterEventInput!):String!
+    newClusterAction(action:ClusterActionInput!):String!
+}
+
+type Subscription{
+    clusterEventListener(project_id:String!):ClusterEvent!
```

Comment:
This is subscription is not part of cluster register. You can remove it. Same with ClusterEvent response

Reply:
Can we keep this one? Because we won't be able to test the code unless we add this again in a separate PR, we already have 2 PRs open. I think it might not be necessary to remove this as it's just one small function. I'll add comments to make it easier to understand.

Reply:
I need this one to work on the argo logs.